nsdb-cluster/src/main/scala/io/radicalbit/nsdb/cluster/actor/ClusterListener.scala

```diff
@@ -87,28 +95,56 @@ class ClusterListener() extends Actor with ActorLogging {
 
   override def postStop(): Unit = cluster.unsubscribe(self)
 
+  protected def createNodeActorsGuardian(): ActorRef =
+    context.system.actorOf(
+      NodeActorsGuardian.props(self).withDeploy(Deploy(scope = RemoteScope(cluster.selfMember.address))),
+      name = s"guardian_$selfNodeName"
+    )
+
+  protected def retrieveLocationsToAdd: List[LocationWithCoordinates] =
+    FileUtils.getLocationsFromFilesystem(indexPath, selfNodeName)
+
+  protected def onSuccessBehaviour(readCoordinator: ActorRef,
+                                   writeCoordinator: ActorRef,
+                                   metadataCoordinator: ActorRef,
+                                   publisherActor: ActorRef): Unit =
+    new NsdbNodeEndpoint(readCoordinator, writeCoordinator, metadataCoordinator, publisherActor)(context.system)
+
+  protected def onFailureBehaviour(member: Member, error: Any): Unit = {
+    log.error("received wrong response {}", error)
+    cluster.leave(member.address)
+  }
+
+  protected def selectNodeActorsGuardian: ActorSelection =
```

Comment:
maybe this is redundant. If we create the actor guardian making sure to use the same name inside the main and the test class, it wouldn't be necessary to redefine it in the test class, hence to have this method whatsoever